qiskit_experiments/calibration_management/basis_gate_library.py

```diff
@@ -289,3 +290,95 @@ def default_values(self) -> List[DefaultCalValue]:
+        rotaries: bool = True,
+    ):
+        """Setup the library.
+
+        Args:
+            basis_gates: The basis gates to generate.
+            default_values: A dictionary to override library default parameter values.
+            rotaries: If True (the default) then drives will be added to the target qubit
+                during the CR tones on the control qubit.
+        """
+        self._rotaries = rotaries
+        super().__init__(basis_gates, default_values)
+
+    @property
+    def __supported_gates__(self) -> Dict[str, int]:
+        """The supported gates of the library are two-qubit pulses for the ecr gate."""
+        return {"cr45p": 2, "cr45m": 2, "ecr": 2}
+
+    def default_values(self) -> List[DefaultCalValue]:
+        """The default values of the CR library."""
+        defaults = []
+        for name, schedule in self.items():
+            for param in schedule.parameters:
+                if "ch" not in param.name:
+                    value = self._default_values[param.name]
+                    defaults.append(DefaultCalValue(value, param.name, tuple(), name))
+
+        return defaults
+
+    def _build_schedules(self, basis_gates: Set[str]) -> Dict[str, ScheduleBlock]:
+        """Build the schedules of the CR library."""
+
+        schedules = dict()
+
+        rot_amp = Parameter("tamp")
+        sigma = Parameter("σ")
+        cr_amp = Parameter("amp")
+        cr_dur = Parameter("duration")
+        cr_width = Parameter("w")
+        t_chan_idx = Parameter("ch1")
+        u_chan_idx = Parameter("ch0.1")
+        t_chan = pulse.DriveChannel(t_chan_idx)
+        u_chan = pulse.ControlChannel(u_chan_idx)
+
+        if "cr45p" in basis_gates:
+            with pulse.build(name="cr45p") as cr45p:
+                pulse.play(
+                    pulse.GaussianSquare(cr_dur, cr_amp, width=cr_width, sigma=sigma), u_chan
+                )
+
+                if self._rotaries:
+                    pulse.play(
+                        pulse.GaussianSquare(cr_dur, rot_amp, width=cr_width, sigma=sigma), t_chan
+                    )
+
+            schedules["cr45p"] = cr45p
+
+        if "cr45m" in basis_gates:
+            with pulse.build(name="cr45m") as cr45m:
+                pulse.play(
+                    pulse.GaussianSquare(cr_dur, -cr_amp, width=cr_width, sigma=sigma), u_chan
+                )
+
+                if self._rotaries:
+                    pulse.play(
+                        pulse.GaussianSquare(cr_dur, -rot_amp, width=cr_width, sigma=sigma), t_chan
+                    )
+
+            schedules["cr45m"] = cr45m
+
+        if "ecr" in basis_gates:
+            with pulse.build(name="ecr") as ecr:
+                with pulse.align_sequential():
+                    pulse.reference("cr45p", "q0", "q1")
+                    pulse.reference("x", "q0")
+                    pulse.reference("cr45m", "q0", "q1")
+                    pulse.reference("x", "q0")
```

Comment:
Where do you take the `ecr` definition from? I have always used `ecr` as this schedule without the final `x`. That seems to be what is documeanted for `ECRGate` here: https://qiskit.org/documentation/stubs/qiskit.circuit.library.ECRGate.html.

Reply:
Good point. Done here: [102617f](https://github.com/Qiskit/qiskit-experiments/pull/889/commits/102617f481457efaa958082b0a7f2c2ca26fee75)